### ISSUE TITLE
Constant mpris volume propery updates

### ISSUE THREAD

**Issue by saivert** [quodlibet/quodlibet#2362]
I notice mpris constantly emits PropertiesChanged for Volume.

According to dbus-monitor:
```
signal time=1492641745.059240 sender=:1.1077 -> destination=(null destination) serial=7232 path=/org/mpris/MediaPlayer2; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "org.mpris.MediaPlayer2.Player"
   array [
      dict entry(
         string "Volume"
         variant             double 1
      )
   ]
   array [
   ]
```

This causes skips in the sound output because volume is intermittently adjusted.

**Comment by lazka** [comment]
I can't reproduce here. What mpris client are you using? Maybe it's changing our volume...

**Comment by saivert** [comment]
The client I'm using is: https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer
I can't repro this with deadbeef's mpris plugin.

**Comment by lazka** [comment]
thanks, will give it a try.

**Comment by lazka** [comment]
I can't reproduce.

Quod Libet currently emits volume changes multiple times when seeking and when changing songs due to a workaround for a bug in the GStreamer pulsesink element (it does not emit volume changes properly, so we poll at certain points). This doesn't mean that volume gets changed.

I'm out of ideas atm.

**Comment by saivert** [comment]
Maybe do a check if the volume value actually changed before emitting PropertiesChanged? I dunno. I do hear a faint skip (as if there is distortion in the sound) just as dbus-monitor outputs the event.
I will do some more testing later tonight.

**Comment by lazka** [comment]
Yeah, we could do that I guess if we mirror the pulseaudio volume state.

Do you use pulseaudio?

**Comment by saivert** [comment]
Yes I use pulseaudio.

**Comment by lazka** [comment]
If you run "pactl subscribe" you should see "Event 'change' on sink-input..." for ever volume change. Is that the case whenever you hear skips?

And please check if the problem persists with MPRIS enabled and the gnome-shell plugin disabled.

thanks

**Comment by saivert** [comment]
ok. This only happens when playing an internet radio stream. Not local files.
Try this stream: http://sc-atr.1.fm:7700/

**Comment by lazka** [comment]
Ah, try to increase the buffer duration in the preferences (under "playback"). The short skips are when the buffer underruns and QL has to rebuffer. I don't know what can influence flakyness in that area, but I get lots of underruns as well here for this stream.

And a "rebuffer" action triggers an event which we use to emit volume changes for the above mentioned bug workaround.

**Comment by saivert** [comment]
It helps. 4 seconds and the problem is gone.
So the skips are actually caused by the buffer underrun because the volume level stays the same.
I guess this would have been revealed earlier if I had some underrun reporting.

**Comment by lazka** [comment]
We do show a buffer status progress bar in the bottom right corner, but the rebuffer duration is so small that it's not visible long enough. Definitely something that needs improvement.
